app/components/machine/driver-aliyunecs/component.js

```diff
@@ -28,5 +28,52 @@ export default Ember.Component.extend(Driver, {
 
     //this.set('editing', false);
   },
+  validate: function(){
+    this._super();
+    var errors = this.get('errors') || [];
+    var name = this.get('model.hostname');
+    if (name) {
+      //name length rules
+      if (name.length<2){
+        errors.push("Name should have more then 2 letters.");
+      }
+      //check '.','-' can not be used as the first letter and the last letter
+      if (!/^[a-zA-Z0-9]([\.-]?[a-zA-Z0-9]+[\.-]?)*[a-zA-Z0-9]$/.test(name)){
+        errors.push("Name doesn't match the rule of Aliyunecs hostname rule");
+      }
+      //windows hostname can not have more then 15 letter
 
+      //linux hostname can not have more then 30 letter
+      if (name.length>30){
+        errors.push("The hostname can not have more then 30 letters.");
+      }
+    }
+    var accessKey = this.get('aliyunecsConfig.accessKeyId');
+    var accessSecret = this.get('aliyunecsConfig.accessKeySecret');
+    if (!accessKey){
+      errors.push("Access key is required.");
+    }
+    if (!accessSecret){
+      errors.push("Access Secret is required");
+    }
+    var sshPassword = this.get('aliyunecsConfig.sshPassword');
+    if (!sshPassword){
+      errors.push("SSH Password is required");
+    }
+    if (sshPassword && (sshPassword.length<8) || sshPassword.length>30){
+      errors.push("SSH Password should have at least 8 chatacter and less then 30 chatacter");
+    }
+    if (!/[?+*$^().|<>';:\-=\[\]\{\},&%#@!~`\\a-zA-Z0-9]+/.test(sshPassword)) {
```

Comment:
We're completely following the rules of Aliyun ECS right?

Reply:
Yes, completely.